packages/website/docs/_samples/ai/TextArea/Extended/main.js

```diff
@@ -0,0 +1,369 @@
+
+	if (!menuItem) {
+		return;
+	}
+
+	const processingLabel = menuItem.dataset.processingLabel || 'Processing...';
+	const textKey = menuItem.dataset.textKey || 'en';
+
+	saveCurrentVersion();
+	currentActionInProgress = action;
+	currentGenerationIndex += 1;
+	const generationIdForThisRun = currentGenerationIndex;
+
+	setLoadingState(processingLabel);
+
+	// Replace this delay with actual API call
+	await delay(TIMING_CONFIG.processingDelay);
+
+	if (!textarea.loading || generationIdForThisRun !== currentGenerationIndex) {
+		resetGenerationState();
+		return;
+	}
+
+	const text = SAMPLE_TEXTS[textKey] || SAMPLE_TEXTS.en;
+	await animateTextGeneration(text, action, menuItem);
+}
+
+function stopGeneration() {
+	if (!textarea.loading) {
+		return;
+	}
+
+	stopTypingAnimation();
+	currentGenerationIndex += 1;
+	const action = currentActionInProgress || 'generate';
+	const menuItem = findMenuItemByAction(action);
+	const completedLabel = (menuItem && menuItem.dataset.completedLabel) ? menuItem.dataset.completedLabel : 'Action completed';
+
+	versionHistory.push({
+		value: textarea.value,
+		action,
+		endAction: completedLabel + " (stopped)",
+		timestamp: new Date().toISOString()
+	});
+
+	currentIndexHistory = versionHistory.length - 1;
+	currentActionInProgress = null;
+
+	buildMenuFromConfig();
+	updateComponentState();
+	textarea.loading = false;
+}
+
+function handleVersionChange(event) {
+	const { backwards } = event.detail || {};
+
+	if (backwards && currentIndexHistory > 0) {
+		saveCurrentVersion();
+		updateComponentState(currentIndexHistory - 1);
+	} else if (!backwards && currentIndexHistory < versionHistory.length - 1) {
+		saveCurrentVersion();
+		updateComponentState(currentIndexHistory + 1);
+	}
+}
+
+async function handleMenuItemClick(event) {
+	const action = event?.detail?.item?.dataset?.action;
+
+	if (!action) {
+		return;
+	}
+
+	await executeAction(action);
+}
+
+function handleStopGeneration() {
+	stopGeneration();
+}
+
+function handleKeydown(event) {
+	if (event.key === 'Escape' && textarea.loading) {
+		event.preventDefault();
+		stopGeneration();
+	}
+}
+
+function resetDemo() {
+	stopGeneration();
+	
+	versionHistory = [];
+	currentIndexHistory = 0;
+	currentGenerationIndex = 0;
+	currentActionInProgress = null;
+	
+	textarea.value = "Innovation managers operate with both creativity and business acumen, driving initiatives that cultivate an innovation-friendly culture.";
+	textarea.loading = false;
+	textarea.actionText = '';
+	textarea.currentVersionIndex = 1;
+	textarea.totalVersions = 1;
+
+	buildMenuFromConfig();
+}
+
+function toggleReadonly() {
+	textarea.readonly = !textarea.readonly;
+	readonlyToggle.textContent = textarea.readonly ? 'Enable Editing' : 'Toggle Readonly';
+}
+
+function toggleDisabled() {
+	textarea.disabled = !textarea.disabled;
+	disabledToggle.textContent = textarea.disabled ? 'Enable Component' : 'Toggle Disabled';
+}
+
+function initDemo() {
```

Comment:
IMO event listeneres could be added after constants, but if you prefer to have method, maybe we can rename it to just "init()"